### PR TITLE
[Core] Placement Group: Fix Flakey Test placement_group_test_5 and Typo

### PR DESCRIPTION
**Pull request by jon-chuang**
## Why are these changes needed?

placement_group_test_5 is flakey. Reason is requesting PG with exact object store memory as node. If object store has object, then PG scheduling fails.

Also fix bug - typo. 

## Related issue number
Introduced by: https://github.com/ray-project/ray/pull/22762

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_resource_scheduler.cc [hunk 1]
```diff
@@ -222,7 +222,7 @@ std::vector<int> GcsResourceScheduler::SortRequiredResources(
     for (auto r : extra_resources_set) {
       auto a_iter = a.custom_resources.find(r);
       const auto &a_resource = a_iter != a.custom_resources.end() ? a_iter->second : 0;
-      auto b_iter = a.custom_resources.find(r);
```
Comment (jjyao) [inline src/ray/gcs/gcs_server/gcs_resource_scheduler.cc:225]: no test caught this? If so, can we add one?

**Review by jjyao — DISMISSED** [review]

File: src/ray/gcs/gcs_server/gcs_resource_scheduler.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/gcs/gcs_server/gcs_resource_scheduler.cc:225]: +1. Can you add a cpp test for this case?

Comment (jon-chuang) [reply src/ray/gcs/gcs_server/gcs_resource_scheduler.cc:225]: Yep, will do, previously would have been caught by object store memory, but was not caught due to resource request refactoring!

Comment (jon-chuang) [reply src/ray/gcs/gcs_server/gcs_resource_scheduler.cc:225]: added custom resource which causes test to fail prior to this PR

**Comment by jon-chuang** [comment]
I think test failures are unrelated

**Comment by rkooo567** [comment]
Hmm can you merge the latest master again? It is a bit weird cpp asan test failed (I've never seen it failed)

**Comment by jon-chuang** [comment]
seems the tests pass now

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**